### PR TITLE
[kernel] Copy the kernel config

### PR DESCRIPTION
**Pull request by aadhar-agarwal**
Copy the kernel config - `/boot/config-{kernel-version}`

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/kernel.py [hunk 1]
```diff
@@ -147,4 +148,12 @@ def setup(self):
         if not self.get_option("trace"):
             self.add_forbidden_path("/sys/kernel/debug/tracing/trace")
 
+
+class AzureKernel(Kernel, AzurePlugin):
```
Comment (TurboTurtle) [inline sos/report/plugins/kernel.py:152]: The way the tagging resolution works out, we'd have to change `IndependentPlugin` to the specific `RedHatPlugin`, `UbuntuPlugin`, etc... tags if we break out one for just Azure Linux here.

That said......how have we not been collecting this generally for so long? @pmoravec @arif-ali should this become a general collection? I'm surprised that we don't do this today already, but I have to assume that no one has been asking for it until now...somehow?

I think we'd probably want this to go into the regular collection, and keep `Kernel` as a strictly `IndependentPlugin`.

Comment (aadhar-agarwal) [reply sos/report/plugins/kernel.py:152]: Ah I see, the `IndependentPlugin` are for plugins that are distribution agnostic?

Happy to move this to the regular collection (the kernel setup function) above and keep `Kernel` as a strictly `IndependentPlugin`

Comment (arif-ali) [reply sos/report/plugins/kernel.py:152]: Agreed, that could be useful for Ubuntu Plugin as well, so agree to add this above in the main class

Comment (aadhar-agarwal) [reply sos/report/plugins/kernel.py]: @TurboTurtle @arif-ali, I added `/boot/config-{kernel-version}` to the main class above and removed the AzureKernel specific class

File: sos/report/plugins/kernel.py [hunk 2]
```diff
@@ -136,7 +136,8 @@ def setup(self):
             "/sys/kernel/debug/extfrag/extfrag_index",
             clocksource_path + "available_clocksource",
             clocksource_path + "current_clocksource",
-            "/proc/pressure/"
+            "/proc/pressure/",
+            "/boot/config-%s" % self.policy.kernel_version()
```
Comment (TurboTurtle) [inline sos/report/plugins/kernel.py:140]: Please use an f-string here as we are converting the project to use those globally.

Comment (aadhar-agarwal) [reply sos/report/plugins/kernel.py:140]: For sure! Can I also change [line 114](https://github.com/sosreport/sos/pull/3580/files#diff-622aa52be019af33ea587ea315ab1be1333530c98581290b57c6e44e493bae2cR114) to use an f string as well?

Comment (arif-ali) [reply sos/report/plugins/kernel.py:140]: > For sure! Can I also change [line 114](https://github.com/sosreport/sos/pull/3580/files#diff-622aa52be019af33ea587ea315ab1be1333530c98581290b57c6e44e493bae2cR114) to use an f string as well?

you sure can :)

Comment (aadhar-agarwal) [reply sos/report/plugins/kernel.py]: Updated to use f-strings! :)

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]

**Review by arif-ali — APPROVED** [review]

**Pull request merged**